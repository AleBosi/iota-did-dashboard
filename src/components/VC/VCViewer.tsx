import React from "react";

interface VCData {
  subject: string;
  type: string;
  value: string;
  issuedAt?: string;
  issuer?: string;
}

interface Props {
  vc: VCData;
}

export default function VCViewer({ vc }: Props) {
  return (
    <div className="border rounded p-4 bg-gray-50 mb-2">
      <div><b>Subject:</b> {vc.subject}</div>
      <div><b>Tipo:</b> {vc.type}</div>
      <div><b>Valore:</b> {vc.value}</div>
      {vc.issuer && <div><b>Issuer:</b> {vc.issuer}</div>}
      {vc.issuedAt && <div><b>Emessa il:</b> {vc.issuedAt}</div>}
    </div>
  );
}
