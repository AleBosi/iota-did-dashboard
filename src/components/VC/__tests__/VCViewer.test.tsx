import { render, screen } from "@testing-library/react";
import VCViewer from "../VCViewer";

const baseVc = {
  subject: "did:iota:123",
  type: "Compliance",
  value: "Certificato OK"
};

describe("VCViewer component", () => {
  it("visualizza subject, type e value", () => {
    render(<VCViewer vc={baseVc} />);
    expect(screen.getByText(baseVc.subject)).toBeInTheDocument();
    expect(screen.getByText(baseVc.type)).toBeInTheDocument();
    expect(screen.getByText(baseVc.value)).toBeInTheDocument();
  });

  it("visualizza issuer se presente", () => {
    render(<VCViewer vc={{ ...baseVc, issuer: "did:iota:issuer" }} />);
    expect(screen.getByText("did:iota:issuer")).toBeInTheDocument();
  });

  it("NON visualizza issuer se non presente", () => {
    render(<VCViewer vc={baseVc} />);
    expect(screen.queryByText(/did:iota:issuer/i)).toBeNull();
  });

  it("visualizza issuedAt se presente", () => {
    render(<VCViewer vc={{ ...baseVc, issuedAt: "2025-08-05" }} />);
    expect(screen.getByText("2025-08-05")).toBeInTheDocument();
  });

  it("NON visualizza issuedAt se non presente", () => {
    render(<VCViewer vc={baseVc} />);
    expect(screen.queryByText(/2025-08-05/i)).toBeNull();
  });

  it("matcha lo snapshot", () => {
    const { asFragment } = render(
      <VCViewer vc={{ ...baseVc, issuer: "did:iota:issuer", issuedAt: "2025-08-05" }} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
