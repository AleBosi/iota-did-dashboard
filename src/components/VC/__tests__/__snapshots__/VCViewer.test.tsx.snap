// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`VCViewer component matcha lo snapshot 1`] = `
<DocumentFragment>
  <div
    class="border rounded p-4 bg-gray-50 mb-2"
  >
    <div>
      <b>
        Subject:
      </b>
       did:iota:123
    </div>
    <div>
      <b>
        Tipo:
      </b>
       Compliance
    </div>
    <div>
      <b>
        Valore:
      </b>
       Certificato OK
    </div>
    <div>
      <b>
        Issuer:
      </b>
       did:iota:issuer
    </div>
    <div>
      <b>
        Emessa il:
      </b>
       2025-08-05
    </div>
  </div>
</DocumentFragment>
`;
